contributing.md

```diff
@@ -0,0 +1,153 @@
+    
+    |Key| Type                                                 |Required| Description                                                                                                                                                                          |
+    |-|------------------------------------------------------|-|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+    |`title`| string                                               |Yes| The title of the troubleshooting entry, displayed as a heading above it                                                                                                              |
+    |`section`| The literal string `troubleshooting`                 |Yes| Must be `troubleshooting`, used to identify troubleshooting entries during site build                                                                                                |
+    |`products` or `topics`| array of strings                                     |Yes (can have either or both, but must have at least one)| The products or topics related to the entry. The entry will show up on the troubleshooting pages for the listed products and topics.                                                 |
+    |`errors`| object of form `{language: string, message: string}` |No| The error, if any, related to the troubleshooting entry. Displayed as a code block right underneath the title. `language` is the programming language to use for syntax highlighting. |
+    |`keywords`| array of strings                                     |No| These are displayed at the bottom of every troubleshooting page. Each keyword links to a collection of all pages associated with that keyword.                                       |
+    |`tags`| array of strings                                     |No| Concepts, actions, or things associated with the troubleshooting entry. These are not displayed in the UI, but they affect the calculation of related pages.                         |
+    
+    Beneath the front matter, describe the error and its solution in regular Markdown. You can also use any other components allowed within the docs site.
+    
+    The entry shows up on the troubleshooting pages for its associated products and topics. If the page doesn't already exist, add an entry for it in the page
+    index, setting `type` to `placeholder`. See [Navigation tree](#navigation-tree).
+
+#### Navigation tree
+
+The navigation hierarchy of a doc section is governed by `page-index/page-index.js` within the corresponding directory. To change the structure, for example, add or delete pages in a section, modify the corresponding `page-index.js`.
```

Comment:
Add a line explaining what the structure does. Order and visibility of files etc. 

For example is always at the start of a sentence.